SWLOR.Game.Server/Feature/GuiDefinition/ViewModel/HoloNetViewModel.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Threading.Tasks;
+using Discord;
+using Discord.Webhook;
+using SWLOR.Game.Server.Service;
+using SWLOR.Game.Server.Service.GuiService;
+
+namespace SWLOR.Game.Server.Feature.GuiDefinition.ViewModel
+{
+    public class HoloNetViewModel : GuiViewModelBase<HoloNetViewModel, GuiPayloadBase>
+    {
+        public string HoloNetText
+        {
+            get => Get<string>();
+            set => Set(value);
+        }
+
+        public const int MaxHoloNetTextLength = 800;
+        private const int Price = 2500;
+
+        protected override void Initialize(GuiPayloadBase initialPayload)
+        {
+            HoloNetText = string.Empty;
+            WatchOnClient(model => model.HoloNetText);
+        }
+
+        public Action OnClickSubmit() => () =>
+        {
+            if (string.IsNullOrWhiteSpace(HoloNetText))
+            {
+                return;
+            }
+
+            var message = HoloNetText;
+
+            if (message.Length > 600)
+            {
+                SendMessageToPC(Player, "Your HoloNet broadcast was too long. Please shorten it to no longer than 600 characters and resubmit the broadcast. For reference, your message was: \"" + message + "\"");
+                return;
+            }
+
+            var url = Environment.GetEnvironmentVariable("SWLOR_HOLONET_WEBHOOK_URL");
```

Comment:
I would put a modal "Are you sure?" confirmation around the rest of this method. This will prevent players from accidentally hitting submit before they're ready.

Example:
```


            ShowModal("Are you sure you want to submit this broadcast?", () =>
            {
                      var url = Environment.GetEnvironmentVariable(...);
                      <rest of code>
            });
```

Reply:
Modal'd